### ISSUE TITLE
TLSv1.3 exporter secret is not correct

### ISSUE THREAD

**Issue by tatsuhiro-t** [openssl/openssl#3940]
Current code always passes empty data to tls13_hkdf_expand.
But according to https://tlswg.github.io/tls13-spec/draft-ietf-tls-tls13.html#rfc.section.7.1, it should be a hash value of empty data.

**Comment by mattcaswell** [comment]
That is not the way I read the spec:

    Note that in some cases a zero-length HashValue (indicated by “”) is passed to HKDF-Expand-Label.

So "" mean a *zero-length* hash value. Not a full length hash value of nothing.

**Comment by mattcaswell** [comment]
Note: this is not the only instance of this. Derive-Secret(x, x, "")  is used in a number of other places. Interoperability testing with other implementations suggest they have the same interpretation.

**Comment by tatsuhiro-t** [comment]
I'm not export in reading TLSv1.3 draft, and could be wrong, but the zero-length HashValue is specifically pointing to the following construct:

```
finished_key =
    HKDF-Expand-Label(BaseKey, "finished", "", Hash.length)
```

See "" is directly passed to HKDF-Expand-Label.

On the other hand, we are talking about `Derive-Secret(Secret, label, "")`.
By definition:
```
    Derive-Secret(Secret, Label, "") =
         HKDF-Expand-Label(Secret, Label,
                           Transcript-Hash(""), Hash.length)

```

This is not the same as `HKDF-Expand-Label(Secret, Label, "", Hash.length)`.

**Comment by tatsuhiro-t** [comment]
Actually, with the current openssl master, I have interop failure with TLS exporter from mint https://github.com/ekr/mint during IETF QUIC hackathon.

**Comment by mattcaswell** [comment]
Yes. On reading it again, I think you are correct.

**Comment by mattcaswell** [comment]
Fixed by c8b93876f1.
